src/overlays/actors/ovl_En_Kusa/z_en_kusa.c

```diff
@@ -7,72 +16,485 @@
+        effectPosition.y = this->actor.posRot.pos.y + (new_var->y * this->actor.scale.y * 20.0f) + 10.0f;
+        effectPosition.z = this->actor.posRot.pos.z + (new_var->z * this->actor.scale.z * 20.0f);
+        fragmentVelocity.x = (Math_Rand_ZeroOne() - 0.5f) * 8.0f;
+        fragmentVelocity.y = Math_Rand_ZeroOne() * 10.0f;
+        fragmentVelocity.z = (Math_Rand_ZeroOne() - 0.5f) * 8.0f;
+        index = (s32)(Math_Rand_ZeroOne() * 111.1f) & 7;
+        EffectSsKakera_Spawn(globalCtx, &effectPosition, &fragmentVelocity, &effectPosition, -100, 64, 40, 3, 0,
+                             D_80A9C26C[index], 0, 0, 80, KAKERA_COLOR_NONE, OBJECT_GAMEPLAY_KEEP, D_040355E0);
+        effectPosition.x = this->actor.posRot.pos.x + (new_var->x * this->actor.scale.x * 40.0f);
+        effectPosition.y = this->actor.posRot.pos.y + (new_var->y * this->actor.scale.y * 40.0f) + 10.0f;
+        effectPosition.z = this->actor.posRot.pos.z + (new_var->z * this->actor.scale.z * 40.0f);
+        fragmentVelocity.x = (Math_Rand_ZeroOne() - 0.5f) * 6.0f;
+        fragmentVelocity.y = Math_Rand_ZeroOne() * 10.0f;
+        fragmentVelocity.z = (Math_Rand_ZeroOne() - 0.5f) * 6.0f;
+        index = (s32)(Math_Rand_ZeroOne() * 111.1f) % 7;
+        EffectSsKakera_Spawn(globalCtx, &effectPosition, &fragmentVelocity, &effectPosition, -100, 64, 40, 3, 0,
+                             D_80A9C26C[index], 0, 0, 80, KAKERA_COLOR_NONE, OBJECT_GAMEPLAY_KEEP, D_040356A0);
+    }
+}
+
+// Spawn Bugs
+void func_80A9B574(EnKusa* this, GlobalContext* globalCtx) {
+    s32 i;
+
+    for (i = 0; i < 3; i++) {
+        Actor* bug =
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_INSECT, this->actor.posRot.pos.x,
+                        this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, Math_Rand_ZeroOne() * 0xFFFF, 0, 1);
+
+        if (bug == NULL) {
+            break;
+        }
+    }
+}
+
+// EnKusa_InitCollider
+void func_80A9B630(Actor* thisx, GlobalContext* globalCtx) {
+    EnKusa* this = THIS;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &D_80A9C208);
+    Collider_CylinderUpdate(&this->actor, &this->collider);
+}
+
+void EnKusa_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnKusa* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, D_80A9C27C);
+    if (globalCtx->csCtx.state != 0) {
+        this->actor.uncullZoneForward += 1000.0f;
+    }
+    // Init for the ColliderCylinder
+    func_80A9B630(thisx, globalCtx);
+
+    func_80061ED4(&this->actor.colChkInfo, NULL, &D_80A9C234);
+    if (this->actor.shape.rot.y == 0) {
+        // Random value I think?
+        s32 rand = (s32)Math_Rand_ZeroFloat(65536.0f);
+        this->actor.posRot.rot.y = (s16)rand;
+        this->actor.initPosRot.rot.y = (s16)rand;
+        this->actor.shape.rot.y = (s16)rand;
+    }
+
+    if (func_80A9AFAC(this, globalCtx, 0.0f) == 0) {
```

Comment:
```suggestion
    if (!func_80A9AFAC(this, globalCtx, 0.0f)) {
```
Although OoT's dialect of C doesn't have bool, variables that only have 1 or 0 and are only compared as ==0 and !=0 can be treated as booleans.